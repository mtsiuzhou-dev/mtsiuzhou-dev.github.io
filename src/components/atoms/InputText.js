import PropTypes from 'prop-types';
import style from './InputText.module.scss';

const InputText = ({
  id,
  value = '',
  placeholder = '',
  onChange,
  onBlur = null,
  innerRef = null,
}) => (
  <input
    id={id}
    className={style.inputText}
    type="text"
    value={value}
    onChange={onChange}
    onBlur={onBlur}
    ref={innerRef}
    placeholder={placeholder}
    autoComplete="off"
    maxLength="75"
  />
);
InputText.defaultProps = {
  value: '',
  placeholder: '',
  onBlur: null,
  innerRef: null,
};
InputText.propTypes = {
  id: PropTypes.string.isRequired,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  innerRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
  ]),
};
export default InputText;
