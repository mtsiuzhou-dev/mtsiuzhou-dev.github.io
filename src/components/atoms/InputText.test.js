import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import InputText from './InputText';

describe('InputText:', () => {
  const id = 'InputTextId';

  test('renders correctly', () => {
    const inputText = renderer
      .create(<InputText id={id} onChange={jest.fn()} />)
      .toJSON();
    expect(inputText).toMatchSnapshot();
  });

  test('onChange should be called', () => {
    const mockOnChange = jest.fn();
    render(<InputText id={id} onChange={mockOnChange} />);
    userEvent.type(screen.getByRole('textbox'), 'typing');

    expect(mockOnChange).toHaveBeenCalled();
  });

  test('onBlur should be called', () => {
    const mockOnBlur = jest.fn();
    render(<InputText id={id} onChange={jest.fn()} onBlur={mockOnBlur} />);
    userEvent.type(screen.getByRole('textbox'), 'typing');

    expect(mockOnBlur).not.toHaveBeenCalled();
    userEvent.tab();
    expect(mockOnBlur).toHaveBeenCalled();
  });

  test('placeholder should be correct', () => {
    const placeholder = 'placeholder1';
    render(
      <InputText id={id} onChange={jest.fn()} placeholder={placeholder} />
    );

    expect(screen.getByRole('textbox').placeholder).toBe(placeholder);
  });
});
