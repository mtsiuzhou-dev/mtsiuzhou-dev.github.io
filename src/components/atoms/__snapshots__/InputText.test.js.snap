// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText: renders correctly 1`] = `
<input
  autoComplete="off"
  className="inputText"
  id="InputTextId"
  maxLength="75"
  onBlur={null}
  onChange={[MockFunction]}
  placeholder=""
  type="text"
  value=""
/>
`;
